pkg/kubectl/cmd/autoscale.go

```diff
@@ -170,18 +170,13 @@ func (o *AutoscaleOptions) Complete(f cmdutil.Factory, cmd *cobra.Command, args
 		return err
 	}
 
-	o.ToPrinter = func(operation string) (printers.ResourcePrinterFunc, error) {
+	o.ToPrinter = func(operation string) (printers.ResourcePrinter, error) {
 		o.PrintFlags.NamePrintFlags.Operation = operation
 		if o.dryRun {
 			o.PrintFlags.Complete("%s (dry run)")
 		}
 
-		printer, err := o.PrintFlags.ToPrinter()
-		if err != nil {
-			return nil, err
-		}
-
-		return printer.PrintObj, nil
+		return o.PrintFlags.ToPrinter()
```

Comment:
your flags should be `genericclioptions`.  Your printers should be `genericclioptions/printers`.  Don't keep them coupled.

Reply:
I'd even say that `genericlioptions` should be just flags and its processing, printers should not be part of that package. `pkg/kubectl/printers` or `pkg/printers` as was before is ok.

Reply:
Done

Reply:
@soltysh 
> printers should not be part of that package. pkg/kubectl/printers or pkg/printers as was before is ok

Since we moved printflags for the `name`, `json`, and `yaml` printers to `genericclioptions`, it was necessary to move the printers to that package as well, in order to break any dependency from that package on kube/kube